pkg/azuredisk/azuredisk.go

```diff
@@ -344,30 +345,43 @@ func (d *Driver) CreateManagedDisk(ctx context.Context, options *ManagedDiskOpti
 	return diskID, nil
 }
 
-// TThe name must begin with a letter or number, end with a letter, number or underscore,
+// Disk name must begin with a letter or number, end with a letter, number or underscore,
 // and may contain only letters, numbers, underscores, periods, or hyphens.
 //
 // See https://docs.microsoft.com/en-us/rest/api/compute/disks/createorupdate#uri-parameters
-func getValidDiskName(volumeName string) string {
-	diskName := volumeName
-	if len(diskName) > diskNameMaxLength {
-		diskName = diskName[0:diskNameMaxLength]
-		klog.Warningf("since the maximum name length is 80, so it is truncated as (%q)", diskName)
+//
+//
+// Snapshot name must begin with a letter or number, end with a letter, number or underscore,
+// and may contain only letters, numbers, underscores, periods, or hyphens.
+// See https://docs.microsoft.com/en-us/rest/api/compute/snapshots/createorupdate#uri-parameters
+func getValidDiskOrSnapshotName(volumeName, volumeType string) string {
+	validName := volumeName
+	if len(validName) > diskOrSnapshotNameMaxLength {
+		validName = validName[0:diskOrSnapshotNameMaxLength]
+		klog.Warningf("since the maximum name length is %d, so it is truncated as (%q)", diskOrSnapshotNameMaxLength, validName)
 	}
-	if !checkDiskName(diskName) || len(diskName) < diskNameMinLength {
-		// maxLength = 80 - (4 for ".vhd") = 75
-		// todo: get cluster name
-		diskName = util.GenerateVolumeName("pvc-disk", uuid.NewUUID().String(), 76)
-		klog.Warningf("the requested volume name (%q) is invalid, so it is regenerated as (%q)", volumeName, diskName)
+	if !checkDiskOrSnapshotName(validName) || len(validName) < diskOrSnapshotNameMinLength {
+		switch volumeType {
+		case "disk":
```

Comment:
I think append a volume name prefix is better instead of a self-defined `volumeType`